port/unix/omrsock.c

```diff
@@ -326,13 +326,13 @@ get_omr_poll_constant(int16_t osPollConstant)
 	if (OMR_ARE_ANY_BITS_SET(osPollConstant, OS_POLLOUT)) {
 		omrPollConstant |= OMRSOCK_POLLOUT;
 	}
-#if !defined(AIXPPC)
 	if (OMR_ARE_ANY_BITS_SET(osPollConstant, OS_POLLERR)) {
 		omrPollConstant |= OMRSOCK_POLLERR;
 	}
 	if (OMR_ARE_ANY_BITS_SET(osPollConstant, OS_POLLNVAL)) {
 		omrPollConstant |= OMRSOCK_POLLNVAL;
 	}
+#if !defined(AIXPPC)
```

Comment:
The condition should be `defined(OMRSOCK_POLLHUP)`.

Also, in `get_os_poll_constant()`, the code should be updated to handle `OS_POLLERR` and `OS_POLLNVAL` unconditionally.

Reply:
This updates `get_omr_poll_constant()` for `POLLERR` and `POLLNVAL` (which are no longer avoided for AIX with changes to `omrsock.h`); I'm suggesting `get_os_poll_constant()` should be updated to remain consistent. If that's not reasonable, then  I don't see how the changes to `omrsock.h` make sense.

Reply:
The thinking is that while on AIX, those values are still valid as return values (ie the os can still report that those events occurred), it is invalid to provide them as a parameter (you can't ask poll to report them, it just assumes you want them reported)